jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/TransitNodeRoutingPrecomputation.java

```diff
@@ -0,0 +1,1214 @@
+         *
+         * @return locality filter of this transit node routing
+         */
+        public LocalityFiler<V> getLocalityFiler() {
+            return localityFiler;
+        }
+
+
+        /**
+         * Constructs a new instance for the given {@code contractionHierarchy},
+         * {@code transitVertices}, {@code transitVerticesPaths}, {@code voronoiDiagram},
+         * {@code accessVertices} and {@code localityFiler}.
+         *
+         * @param contractionHierarchy contraction hierarchy
+         * @param transitVertices      transit vertices
+         * @param transitVerticesPaths paths between transit vertices
+         * @param voronoiDiagram       Voronoi diagram
+         * @param accessVertices       access vertices
+         * @param localityFiler        locality filter
+         */
+        public TransitNodeRouting(ContractionHierarchy<V, E> contractionHierarchy,
+                                  Set<ContractionVertex<V>> transitVertices,
+                                  ManyToManyShortestPathsAlgorithm.ManyToManyShortestPaths<V, E> transitVerticesPaths,
+                                  VoronoiDiagram<V> voronoiDiagram,
+                                  AccessVertices<V, E> accessVertices, LocalityFiler<V> localityFiler) {
+            this.contractionHierarchy = contractionHierarchy;
+            this.transitVertices = transitVertices;
+            this.transitVerticesPaths = transitVerticesPaths;
+            this.voronoiDiagram = voronoiDiagram;
+            this.localityFiler = localityFiler;
+            this.accessVertices = accessVertices;
+        }
+    }
+
+
+    /**
+     * Voronoi diagram for a graph. Formally each cell in the diagram is defined as
+     * $Vor(v) = \{u ∈ V : ∀w ∈ T$ \ $ \{v\} : \mu(u, v) ≤ \mu(u, w)\}$, where $V$ is the vertex set,
+     * $T$ is a set of vertices representing Voronoi cells centers and $\mu(u,v)$ denotes distance
+     * between vertices $u$ and $v$.
+     *
+     * @param <V> graph vertex type
+     */
+    public static class VoronoiDiagram<V> {
+        /**
+         * For each vertex in {@code contractionGraph} contains id of its Voronoi
+         * cell, or {@code NO_VORONOI_CELL} if it does not belong to any cell.
+         */
+        private int[] voronoiCells;
+
+        /**
+         * Constructs a new instance for the given {@code voronoiCells}.
+         *
+         * @param voronoiCells Voronoi cells ids
+         */
+        public VoronoiDiagram(int[] voronoiCells) {
+            this.voronoiCells = voronoiCells;
+        }
+
+        /**
+         * Returns Voronoi cell id which corresponds to {@code vertex}.
+         *
+         * @param vertex vertex
+         * @return Voronoi cell id
+         */
+        public int getVoronoiCellId(ContractionVertex<V> vertex) {
+            return voronoiCells[vertex.vertexId];
+        }
+    }
+
+    /**
+     * Search space based locality filter. For every vertex in the {@code contractionGraph}
```

Comment:
Same comment about public class Javadoc. Need to add clear explanation of what this filter does and what it's needed for.